src/mainwindow.cpp

```diff
@@ -210,6 +220,31 @@ MainWindow::MainWindow(QWidget* parent)
            "E.g. show std::string instead of std::basic_string<char, ...>"));
     connect(prettifySymbolsAction, &QAction::toggled, Settings::instance(), &Settings::setPrettifySymbols);
 
+    auto* collapseTemplatesAction = ui->viewMenu->addAction(tr("Collapse Templates"));
+    collapseTemplatesAction->setCheckable(true);
+    collapseTemplatesAction->setChecked(Settings::instance()->collapseTemplates());
+    collapseTemplatesAction->setToolTip(
+        tr("Collapse complex templates to simpler ones E.g. QHash<...> instead of QHash<QString, QVector<QString>>"));
+    connect(collapseTemplatesAction, &QAction::toggled, Settings::instance(), &Settings::setCollapseTemplates);
+
+    {
+        QWidgetAction* action = new QWidgetAction(this);
+        QWidget* widget = new QWidget(this);
+        QHBoxLayout* layout = new QHBoxLayout();
+        QLabel* label = new QLabel(QLatin1String("Collapse Depth")); // bug fixed here, pointer was missing
```

Comment:
? what does this comment mean? remove it again?

Reply:
also missing `tr()` for the string

Reply:
also, pass `widget` as parent here

Reply:
copied that code from stack overflow and forgot to remove that comment